src/main/java/frc/robot/auton/util/AutonBase.java

```diff
@@ -0,0 +1,53 @@
+package frc.robot.auton.util;
+
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.InstantCommand;
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+
+public abstract class AutonBase extends SequentialCommandGroup {
+    private double m_autonStartTime = 0;
+
+  public AutonBase() {
+    String name = this.getClass().getSimpleName();
+    name = name.substring(name.lastIndexOf('.') + 1);
+
+    AutoChooser.addAuton(this, name);
+  }
+
+  public AutonBase(String name) {
+    AutoChooser.addAuton(this, name);
+  }
+
+
+// TODO: Possibly look into a low priority issue for changing printing to logging 
+  public void addCommandsWithLog(String tag, Command... commands) {
+    // Add Printing to each command
+
+    // Print start of group
+    super.addCommands(
+        new InstantCommand(
+            () -> {
+              m_autonStartTime = Timer.getFPGATimestamp();
+              System.out.println("Auton group Starting at {}" + m_autonStartTime);
```

Comment:
If you want the time in the brackets, you'll need to change this to '"Auton group Starting at {" + m_autonStartTime + "}"'

And the same for all the future logs too.